src/irmin/conf.mli

```diff
@@ -83,27 +98,30 @@ val docv : 'a key -> string option
 val docs : 'a key -> string option
 (** [docs k] is [k]'s documentation section (if any). *)
 
-val root : string option key
+val root : Spec.t -> string key
 (** Default [--root=ROOT] argument. *)
 
 (** {1:conf Configurations} *)
 
 type t
 (** The type for configurations. *)
 
-val empty : t
-(** [empty] is the empty configuration. *)
+val spec : t -> Spec.t
+(** [spec c] is the specification associated with [c] *)
 
-val singleton : 'a key -> 'a -> t
-(** [singleton k v] is the configuration where [k] maps to [v]. *)
+val empty : Spec.t -> t
+(** [empty spec] is an empty configuration. *)
+
+val singleton : Spec.t -> 'a key -> 'a -> t
+(** [singleton spec k v] is the configuration where [k] maps to [v]. *)
 
 val is_empty : t -> bool
 (** [is_empty c] is [true] iff [c] is empty. *)
 
 val mem : t -> 'a key -> bool
 (** [mem c k] is [true] iff [k] has a mapping in [c]. *)
 
-val add : t -> 'a key -> 'a -> t
+val add : t -> ?verify:bool -> 'a key -> 'a -> t
 (** [add c k v] is [c] with [k] mapping to [v]. *)
```

Comment:
can you also document the `verify` arg?

Reply:
I removed the verify argument since it was kind of a hack anyway.